src/test/java/dao/ProductAccountDaoImplTest.java

```diff
@@ -0,0 +1,62 @@
+package dao;
+
+import db.Storage;
+import model.ProductAccount;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ProductAccountDaoImplTest {
+
+    private Storage memdb;
+    private ProductAccountDaoImpl dao;
+    private ProductAccount product0;
+    private ProductAccount product1;
+    private ProductAccount product2;
+
+    @Before
+    public void setup() {
+
+        memdb = new Storage();
+        memdb.products.clear();
+        dao = new ProductAccountDaoImpl(memdb);
+        product0 = new ProductAccount("OriginalProduct0").setAmount(13);
+        product1 = new ProductAccount("OriginalProduct1").setAmount(26);
+        product2 = new ProductAccount("OriginalProduct2").setAmount(17);
+        dao.add(product0);
+        dao.add(product1);
+        dao.add(product2);
+    }
+
+    @Test
+    public void testAdd() {
+
+        Assert.assertEquals("Product is equal",product0,memdb.products.get(0));
+        Assert.assertEquals("Product is equal",product1,memdb.products.get(1));
+        Assert.assertEquals("Product is equal",product2,memdb.products.get(2));
```

Comment:
you should write here an error message, not a success message, fix this in all similar cases